### PR TITLE
Parser: Flex Prefix

### PR DESCRIPTION
**Pull request by ax3l**
Add a prefix to all flex symbols in the parser. This ensures that in builds with complex dependencies, which might also use flex for their own purposes, no symbol collisions occur.

Pro-active prevents issues such as: https://github.com/ornladios/ADIOS2/issues/2576

### CODE REVIEW

File: Source/Parser/wp_parser.l [hunk 1]
```diff
@@ -6,6 +6,7 @@
 %}
 
 %option nounput
+%option prefix="wxparser"
```
Comment (ax3l) [inline Source/Parser/wp_parser.l:9]: @WeiqunZhang does the `wp_parser` prefix in the files here stand for `warp`(x)? In that case, I would also make the prefix `wpparser` instead.

**Comment by ax3l** [comment]
I called `make` but it looks like `wp_parser.tab.cpp` needs to be updated, too. Is this file generated?

```
/usr/bin/ld: Source/Parser/wp_parser.tab.cpp.o: in function `yyparse()':
Source/Parser/wp_parser.tab.cpp:1294: undefined reference to `yylex()'
collect2: error: ld returned 1 exit status
```

File: Source/Parser/wp_parser.l (hunk 1 above)
Comment (WeiqunZhang) [reply Source/Parser/wp_parser.l:9]: I think `wp` was chosen to stand for `W`arpX `P`arser.

Comment (WeiqunZhang) [reply Source/Parser/wp_parser.l:9]: `wxparser` is fine.

**Comment by WeiqunZhang** [comment]
```
diff --git a/Source/Parser/wp_parser.y b/Source/Parser/wp_parser.y
index 809dbfa..7366eb4 100644
--- a/Source/Parser/wp_parser.y
+++ b/Source/Parser/wp_parser.y
@@ -4,7 +4,7 @@
     #include <stdlib.h>
     #include <math.h>
     #include "wp_parser_y.h"
-    int yylex (void);
+    int wxparserlex (void);
 %}
 
 /* We do not need to make this reentrant safe, because we use flex and
@@ -12,6 +12,7 @@
    thread safe.
 */
 /*%define api.pure full */
+%define api.prefix {wxparser}
 
 /* This is the type returned by functions wp_new* declared in
    wp_parser_y.h.  See also bison rules at the end of this file.
```

Could you apply this patch and rerun `make`?

**Comment by WeiqunZhang** [comment]
We also need
```
--- a/Source/Parser/wp_parser_c.cpp
+++ b/Source/Parser/wp_parser_c.cpp
@@ -6,7 +6,7 @@ struct wp_parser*
 wp_c_parser_new (char const* body)
 {
     YY_BUFFER_STATE buffer = wxparser_scan_string(body);
-    yyparse();
+    wxparserparse();
     struct wp_parser* parser = wp_parser_new();
     wxparser_delete_buffer(buffer);
     return parser;
```

**Comment by WeiqunZhang** [comment]
Can I push to your branch?  I have a few other fixes.

**Comment by ax3l** [comment]
Yes, will do those!

Does `wp_parser.tab.cpp` need updates as well or is that file unused?

**Comment by WeiqunZhang** [comment]
`wp_parser.tab.cpp` is auto generated.  The `yy` there is fine because of macros like
```
#define yyerror         wxparsererror
```
at the beginning of the file.

**Comment by WeiqunZhang** [comment]
I have a working version now.  I will push the changes in a minute.

**Comment by WeiqunZhang** [comment]
@ax3l Could rerun `make` to generate `wp_parser.lex.cpp`, `wp_parser.lex.h`, `wp_parser.tab.cpp`, and `wp_parser.tab.h`?  My versions of flex and bison are very old.  It would result in too many changes.

**Comment by ax3l** [comment]
Thank you, push incoming :eagle:

**Review by lucafedeli88 — APPROVED** [review]
Thanks for this PR!

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**